nucypher/blockchain/eth/chains.py

```diff
@@ -1,88 +1,74 @@
-import random
-from abc import ABC
-from typing import List
+from typing import List, Union
 
-from nucypher.blockchain.eth.constants import NucypherMinerConfig
-from nucypher.blockchain.eth.interfaces import ContractProvider
+from nucypher.blockchain.eth.constants import NucypherMinerConstants
+from nucypher.blockchain.eth.interfaces import ContractInterface, DeployerInterface
 
 
-class TheBlockchain(ABC):
-    """
-    http://populus.readthedocs.io/en/latest/config.html#chains
-
-    mainnet: Connects to the public ethereum mainnet via geth.
-    ropsten: Connects to the public ethereum ropsten testnet via geth.
-    tester: Uses an ephemeral in-memory chain backed by pyethereum.
-    testrpc: Uses an ephemeral in-memory chain backed by pyethereum.
-    temp: Local private chain whos data directory is removed when the chain is shutdown. Runs via geth.
-    """
+class Blockchain:
```

Comment:
I like this name much more.

Reply:
Currently - I doubt it. In future - perhaps other blockchains? Anyway, too early to think that.

I cannot imagine testnet and mainnet being used at the same time

Reply:
Right.  So, it's clearly singular in this runtime.  And since, unlike most singletons, there exists an authoritative, universal mechanism for describing its contents, I actually think that the word "The" gives meaningful insight to the user in the phrase, `TheBlockchain`.  It's not "A Blockchain" - it's a representation of *The* Blockchain.  A separate process will still need to regard the same details as canonical, given connection to the same blockchain.  There is no way for this object to conceptualize "A blockchain" without also being "the blockchain."